### PR TITLE
Add tutorial to use NVDEC with Stream API

### PR DESCRIPTION
**Pull request by mthrok**
This commit adds tutorial to enable/use NVDEC with Stream API.

https://output.circle-artifacts.com/output/job/19e66a4b-1819-4804-8834-d38e6c80c4fd/artifacts/0/docs/hw_acceleration_tutorial.html

Because the use of NVDEC requires build / install FFmpeg from source,
this tutorial was authored on Google Colab, tailored to its environment.

The tutorial here is the result of the notebook execution, with
the link to the publicly accessible Google Colab notebook.

### CODE REVIEW

File: docs/source/hw_acceleration_tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,1306 @@
...
...GrTRi1CIIY1wnY3w3Z221mPisRNdArBEUOWaoQQqKkZntrRJJsKKsCIQRSiv9f6vC37Vxx4YoLV1y44sIVF74ZF77Vi7f/+7/6V3/kOi5KKcqi5OLinKPj15RlSRAEbG1t0xv02azXFGWzBNNElCeUpQBMDBo7LgwTyzbxkgRWG4LRAC8K2Kw3dDptBsM+nU6bdruF5zVpdnlW4PkhUilevTpmOl1QVwrbbtwcDg4OiKKATbLh4nxCkVdIIel0uihqbCQH+zaen5JtNKZhE4RtbNtFKklVls2TpaxxHZdev0cYOYSxhe9M0XpJp2/R6tVoY4lh5sRhD8cO8f0B6drm5asjJlPN6asp60WBYVoMdgJs22M+M0nXzf/H9VzqqkBWJb7rMbuY4y5L+sqlNRwSdbt8+eVDvvj8U1aLOaUoWK4XRHGb4XAHU9qcvDzHdzp02yP6nS3u3n6HKIiR1Gw2C5SsGPY7dDshpgFZXuD5MWXZPBnHcRe0ZracM0+WbG/1yJOE3d0DHn75iDBs8eLlK7SGXqeHbdq8dfMO+zt9hoOIMAzp9kcslmvcKOTazVuYjg0mZGmCYRoIocnSCs8NsFyfotpQVglJOmU2GzdLXYbC932qMmcxPSdL5mTZkrRIEaKmKnNCN8B3HGBFt7VNLU9Zrc+xzF1acZ9Ou4+oQGsLy3IJ/MYOLS9SpCoIgxa243C4f0iZC/IsY5OsUUrw85/9jL//6c/AdFivV+T5Go2N7QQUWYlluTiOT12VnLx+jaglcdyiKFLKLGWzTBBFlzK/icUhWZpQC7BMm8HIwDBLXN/hxp0OH/5uj6hr8PnHE148Fjh2RLtv4HgJUatCK5uq1qzmgum54LO/S0CGJGmM5/wWjrtFKeZI2cw5XkwqPvmsZDJekKY5RVayWScslxvStKQuNZPzivFpCVqTrCXruY1pOBwebpEXBbWwCAOPO2/f4L3373N4eMhyVWKZHo5jgZK47gTPe4SQkjL3idoOj78qqSuXycUFVVEz2m4Thh6B7xMEPt1eF89zsCyDPLNYrWxW6xZJElHLEMvu4Lghlm1hmI2lnSnNZm5TaSxMoijGwATVKHZlkaG0RDcRkKAuF8m+tm3TaH0ZxqTf+DdeelnXolnKWyyYTiZMLi5YzKfk2QapJAZg2w6uaxIENlFs0+5YxC2DIAKlUjRNyupimb
```
Comment (hwangjeff) [inline docs/source/hw_acceleration_tutorial.ipynb:1275]:
```suggestion
        "We looked at how to build FFmpeg libraries with NVDEC support and use them from TorchAudio. We observe significant speedup when using NVDEC."
```

Comment (hwangjeff) [inline docs/source/hw_acceleration_tutorial.ipynb:1306]: newline

Comment (mthrok) [reply docs/source/hw_acceleration_tutorial.ipynb:1306]: well technically, this file was generated by Google Colab

File: docs/source/hw_acceleration_tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,1306 @@
...
+        "\n",
+        "This tutorial shows how to use Nvidia's hardware video decoding (NVDEC)† with TorchAudio.\n",
+        "\n",
+        "**NOTE**\n",
+        "\n",
+        "> This tutorial is authored in Google Colab, and is tailored for Google Colab's specification.\n",
+        ">\n",
+        "> Please checkout this tutorial in [Google Colab](https://colab.research.google.com/drive/1DDah_IaGULEO66CfQWltRqaVheBkiXdN#sandboxMode=true)\n",
+        ">\n",
+        "> If you apply this in your system, please adjust the configurations accordingly.\n",
```
[the remark this answers is not in the document]
Comment (mthrok) [reply docs/source/hw_acceleration_tutorial.ipynb:16]: The primal concern is the install location of FFmpeg (`/usr/lib`), which is commented in installation section.
Other than that, user's environment can vary, so it's hard to imagine the potential issues.

[the remark this answers is not in the document]
Comment (hwangjeff) [reply docs/source/hw_acceleration_tutorial.ipynb:16]: to be more specific, i am not sure what "apply this in your system" means, e.g. what "this" refers to and what applying it entails

**Review by hwangjeff — APPROVED** [review]

File: docs/source/hw_acceleration_tutorial.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (mthrok) [reply docs/source/hw_acceleration_tutorial.ipynb:16]: ah okay, then let me just rephrase it to `If you install FFmpeg following this tutorial, ...`.
